auth/downscoping/snippets.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Demonstrates how to use Downscoping with Credential Access Boundaries."""
+
+import google.auth
+
+from google.auth import downscoped
+from google.auth.transport import requests
+from google.cloud import storage
+from google.oauth2 import credentials
+
+
+OBJECT_PREFIX_NAME = "customer-a"
+
+
+# [START auth_downscoping_token_broker]
+def get_token_from_broker(bucket_name, object_prefix):
+    """Simulates token broker generating downscoped tokens for specified bucket.
+
+    Args:
+        bucket_name (str): The name of the cloud storage bucket.
+        object_prefix (str): The prefix string of the object blob name. This is used
+            to ensure access is restricted to only objects starting with this
+            prefix string.
+
+    Returns:
+        Tuple[str, datetime]: The downscoped access token and its expiry date.
+    """
+    # [START auth_downscoping_rules]
+    # Initialize the credential access boundary rules.
+    available_resource = f"//storage.googleapis.com/projects/_/buckets/{bucket_name}"
+    # Downscoped credentials will have readonly access to the resource.
+    available_permissions = ["inRole:roles/storage.objectViewer"]
+    availability_expression = (
```

Comment:
Is this going to be presented in the page that explains how to construct these availability expressions? If not, please consider adding a link to the docs that explain how these work.

Reply:
The plan is to present this in the docs page. I have added more comments to help explain each step.